homeassistant/components/rituals_perfume_genie/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""The Rituals Perfume Genie integration."""
+import asyncio
+
+from pyrituals import Account
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_HASH, DOMAIN
+
+EMPTY_CREDENTIALS = ""
+
+PLATFORMS = ["switch"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Rituals Perfume Genie component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Rituals Perfume Genie from a config entry."""
+    session = async_get_clientsession(hass)
+    account = Account(EMPTY_CREDENTIALS, EMPTY_CREDENTIALS, session)
+    account.data = {ACCOUNT_HASH: entry.data.get(ACCOUNT_HASH)}
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = account
+
```

Comment:
We should verify the cloud service is online and they can login here.

`return False` if authentication fails for now and implement reauth in a future PR, https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication

If the cloud service is offline `raise ConfigEntryNotReady`

Reply:
When authenticating with email and password, we receive an `account_hash` wich is always valid. The account_hash is fetched and stored by the config_flow. (We don't use or store the password after receiving the `account_hash`).

The highlighted code recreates the `Account` object and doesn't authenticate. Therefore this will never fail.
How do I check the `account_hash` and if the service is online? Do I have to make a request and catch the exceptions or is there a function in HASS to check if the service is online?

Reply:
If its a single endpoint, we usually setup `DataUpdateCoordinator` and then call `await coordinator.async_refresh()` then raise `ConfigEntryNotReady` if `self.coordinator.last_update_success` is false

https://developers.home-assistant.io/docs/integration_fetching_data/#polling-api-endpoints

Reply:
Commit https://github.com/home-assistant/core/pull/46218/commits/50b993cea028e1a6b04ffd9e044305a1f59c841c adds a connectivity check during the entry setup.